Source/WebInspectorUI/UserInterface/Views/FolderTreeElement.js

```diff
@@ -25,11 +25,36 @@
 
 WI.FolderTreeElement = class FolderTreeElement extends WI.GeneralTreeElement
 {
-    constructor(title, representedObject)
+    constructor(title, representedObject, id = null)
     {
         const classNames = [WI.FolderTreeElement.FolderIconStyleClassName];
         const subtitle = null;
         super(classNames, title, subtitle, representedObject, {hasChildren: true});
+
+        if (id) {
+            let settingName = "folder-tree-element-" + id;
+            const defaultValue = false;
+            this._expandedSetting = new WI.Setting(settingName, defaultValue);
+
+            if (this._expandedSetting.value)
+                this.expand();
+        }
+    }
+
+    collapse()
+    {
+        super.collapse();
+
+        if (this._expandedSetting)
+            this._expandedSetting.value = false;
```

Comment:
NIT: just in case something happens in `super.collapse()` that would change the collapsed state, i'd do this instead
```
this._expandedSetting.value = this.expanded;
```